### PR TITLE
Run setupProviderConfig in BeforeSuite.

### PR DESCRIPTION
**Pull request by spxtr**
If `RunE2ETests` fails or times out on a parallel node then we'll get "Ginkgo timed out waiting for parallel nodes" and no test failures, so we'll call it a success. This was happening when the metadata server requests timed out in `setupProviderConfig`. In this PR, I move that call into `BeforeSuite`, so when it happens we'll mark the test as failed.
#23350

### CODE REVIEW

**Comment by ixdy** [comment]
Isn't this used in some e2e tests, e.g.
https://github.com/kubernetes/kubernetes/blob/0bda969cd5346da74d2a9ce78845d865b6c512f1/test/e2e/pd.go#L528
https://github.com/kubernetes/kubernetes/blob/0bda969cd5346da74d2a9ce78845d865b6c512f1/test/e2e/framework/util.go#L3602

**Comment by spxtr** [comment]
Yes, I'm under the impression that the value of `cloudConfig` is shared across the ginkgo nodes, so initializing it on every node is a waste of time.

**Comment by ixdy** [comment]
`SynchronizedBeforeSuite` only runs on Ginkgo node 1, so `cloudConfig` is only going to be initialized there. The other Ginkgo nodes run in separate processes, so they don't have access to it.

I think I tried initializing it in `SynchronizedBeforeSuite` and then marshaling and passing it to the other nodes, but IIRC `cloudConfig` contains pointers, so that didn't work.

File: test/e2e/e2e.go [hunk 1]
```diff
@@ -135,6 +139,9 @@ var _ = ginkgo.SynchronizedBeforeSuite(func() []byte {
 }, func(data []byte) {
 	// Run on all Ginkgo nodes
 
+	if err := setupProviderConfig(); err != nil {
```
Comment (ixdy) [inline test/e2e/e2e.go]: maybe we don't care, but this is now run *twice* on node 1.

**Comment by ixdy** [comment]
technically it's `SynchronizedBeforeSuite`, not `BeforeSuite` (these are different things), but close enough.

**Comment by spxtr** [comment]
@k8s-bot unit test this issue: #25890

**Pull request merged**